Windows: Simplify build; lowercase lib def name

conda-forge-admin opened the pull request:
This adds small patches to make Windows builds more consistent.
Added myself (@mbargull) to the recipe maintainers to support these changes.

Here's a checklist to do before merging.
- [x] Bump the build number if needed.

Fixes #23

mbargull committed:
Windows: Simplify build; lowercase lib def name

Signed-off-by: Marcel Bargull <marcel.bargull@udo.edu>

mbargull committed:
Add mbargull to recipe-maintainers

Signed-off-by: Marcel Bargull <marcel.bargull@udo.edu>

mbargull commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 24.5.1, conda-smithy 3.36.2, and conda-forge-pinning 2024.07.13.14.05.54

isuruf commented on recipe/bld.bat:
```diff
@@ -1,21 +1,15 @@
+copy libbz2_static.lib %LIBRARY_LIB%\bzip2_static.lib || exit 1
 copy libbz2.lib %LIBRARY_LIB%\bzip2.lib || exit 1
-copy libbz2.dll %LIBRARY_BIN% || exit 1
 copy libbz2.dll %LIBRARY_BIN%\bzip2.dll || exit 1
```
This is not needed right?

mbargull replied:
You mean the `bzip2.dll` specifically, right?
All of those have been grandfathered in from Anaconda's builds (partially, even before `conda.builder` was a thing) and are most likely of limited use.
But yeah, for `bzip2.dll` I can't see any sane reason since the references will all be to `LIBBZ2.dll` anyway.
```suggestion
```

isuruf commented on recipe/meta.yaml:
```diff
@@ -73,3 +74,4 @@ extra:
     - jlanga
```
```suggestion
```

mbargull committed:
Windows: Remove superfluous bzip2.dll

The library's name is LIBBZ2 (or libbz2 after this PR) and as such
LIBBZ2.dll (libbz2.dll) is/will be referenced even if bzip2.lib is used
for linking.

Co-authored-by: Isuru Fernando <isuruf@gmail.com>

Signed-off-by: Marcel Bargull <marcel.bargull@udo.edu>

mbargull committed:
Remove jlanga from recipe-maintainers

They had been added in the initial commit (via staged-recipes) without
comment, plus they are currently not part of conda-forge.

Signed-off-by: Marcel Bargull <marcel.bargull@udo.edu>

mbargull commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 24.5.1, conda-smithy 3.36.2, and conda-forge-pinning 2024.07.13.22.44.53

larsoner commented on recipe/bld.bat:
```diff
@@ -1,21 +1,14 @@
+copy libbz2_static.lib %LIBRARY_LIB%\bzip2_static.lib || exit 1
 copy libbz2.lib %LIBRARY_LIB%\bzip2.lib || exit 1
-copy libbz2.dll %LIBRARY_BIN% || exit 1
-copy libbz2.dll %LIBRARY_BIN%\bzip2.dll || exit 1
```
FYI this removal of `%LIBRARY_BIN\bzip2.dll` creates a bit of inconsistency with other Windows packages, e.g. hdf5, lzo2, lzo1 blosc which are all accessible via the non-`lib-`-prefixed versions in the `%LIBRARY_BIN%`, e.g.:

https://github.com/conda-forge/lzo-feedstock/blob/75ffb613aeaab06b5bef5121f0efcfcb75329a32/recipe/meta.yaml#L34

FWIW this [broke a build I had going over in PyTables](https://github.com/PyTables/PyTables/pull/1183#issuecomment-2229020526). So might be worth reverting this in case other people look specifically for the windows-style naming as well?
